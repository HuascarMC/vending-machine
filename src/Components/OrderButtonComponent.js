import React from 'react';

class OrderButtonComponent extends React.Component {

  pressReturn() {
    this.props.updateChange(this.props.balance);
    this.props.updateBalance(0.00);
  }

  pressPush() {
    this.props.order();
  }

  render() {
    return(
      <div className="order-button-wrapper">
        <button className="order-button" onClick={this.pressReturn.bind(this) }>RETURN</button>
        <button className="order-button">PUSH</button>
      </div>
    );
  }
}

export default OrderButtonComponent;
