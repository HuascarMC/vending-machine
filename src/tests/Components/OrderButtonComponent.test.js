import React from 'react';
import OrderButtonComponent from '../../Components/OrderButtonComponent'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('should render without crashing', () => {
  shallow(<OrderButtonComponent />);
});

it('should render order-button-wrapper div', () => {
  const wrapper = shallow(<OrderButtonComponent />);
  const div = wrapper.find('.order-button-wrapper');
  expect(div.exists()).toEqual(true);
});
