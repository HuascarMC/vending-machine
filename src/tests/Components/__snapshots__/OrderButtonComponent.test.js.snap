// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  className="order-button-wrapper"
>
  <button
    className="order-button"
    onClick={[Function]}
  >
    RETURN
  </button>
  <button
    className="order-button"
    onClick={[Function]}
  >
    PUSH
  </button>
</div>
`;
